providers/keycloak/src/airflow/providers/keycloak/auth_manager/keycloak_auth_manager.py

```diff
@@ -156,12 +215,41 @@ def get_fastapi_app(self) -> FastAPI | None:
                 "This sub application provides login routes."
             ),
         )
-        # authlib requires ``SessionMiddleware``
-        app.add_middleware(
-            SessionMiddleware,
-            secret_key=conf.get("api_auth", "jwt_secret"),
-            https_only=False,
-        )
         app.include_router(login_router)
 
         return app
+
+    def _is_authorized(
+        self,
+        *,
+        method: ResourceMethod | str,
+        resource_type: KeycloakResource,
+        user: KeycloakAuthManagerUser,
+        resource_id: str | None = None,
+    ) -> bool:
+        client_id = conf.get("keycloak_auth_manager", "client_id")
+        realm = conf.get("keycloak_auth_manager", "realm")
+        server_url = conf.get("keycloak_auth_manager", "server_url")
+
+        permission = (
+            f"{resource_type.value}:{resource_id}#{method}"
+            if resource_id
+            else f"{resource_type.value}#{method}"
+        )
+        token_url = f"{server_url}/realms/{realm}/protocol/openid-connect/token"
+        payload = {
+            "grant_type": "urn:ietf:params:oauth:grant-type:uma-ticket",
+            "audience": client_id,
+            "permission": permission,
+        }
+        headers = {
+            "Authorization": f"Bearer {user.access_token}",
+            "Content-Type": "application/x-www-form-urlencoded",
+        }
+        resp = requests.post(token_url, data=payload, headers=headers)
+
+        if resp.status_code == 200:
+            return True
+        if resp.status_code == 403:
+            return False
+        raise AirflowException(f"Unexpected error: {resp.status_code} - {resp.text}")
```

Comment:
How does the user experience differ between these two cases?

Reply:
The first one is something expected. The user does not have access to a given resource. The `is_authorized_` method will return `False`. The second one is unexpected, the api-server raise an error.

I am actually working on a follow-up PR to handle the case where the resource is not defined in Keycloak (which can happen). It should return False in that case.

For all the other use cases I rather raise an error to identity the issue and fix than returning `False` and maybe swallowing an issue without knowing. WDYT?

Reply:
I agree, it makes sense to raise it. Maybe even guide users to check the API logs in the exception logging, where the problem could be more visible there in some cases, rather than what is returned from the API, if it is not related to Keycloak. For example, if the API returns `500`.

Reply:
> The first one is something expected. The user does not have access to a given resource. The is_authorized_ method will return False. The second one is unexpected, the api-server raise an error

Yeah, the code makes sense. But I mean literally what does that look like to the user between those cases. What do they see in the Airflow UI for each case? Or in the logs?

Reply:
> For all the other use cases I rather raise an error to identity the issue and fix than returning False and maybe swallowing an issue without knowing. WDYT?

I'll know for sure after I hear back on the above, but I think I agree!

Reply:
> Yeah, the code makes sense. But I mean literally what does that look like to the user between those cases. What do they see in the Airflow UI for each case? Or in the logs?

If an exception a raised, that will be a 500 on the API server. The user will probably see an error like "Something wrong happened"

Reply:
Interesting, so they'll get a pop-up dialogue of some sort in the UI? To let them know something failed rather than they don't have access? I think that's okay, it should point them to the logs though so they can see more details maybe, or does the traceback show up in the UI dialogue like a dag parse error does?